### PR TITLE
[WIP] backend: Load config schema when backend starts (and optionally validate config)

### PR DESCRIPTION
**Pull request by OrkoHunter**
Related to https://github.com/backstage/backstage/issues/3920

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/backend/src/index.ts [hunk 1]
```diff
@@ -81,6 +83,29 @@ async function main() {
...
+  // loadBackendConfig tells the logger about secrets, so that it can filter out.
+
+  // get all packages
+  const schema = await loadConfigSchema({
+    dependencies: ['@backstage/integration'],
+  });
+  console.log(schema);
+  console.log('serialize');
+
+  const merged = mergeConfigSchemas(
```
Comment (Rugvip) [inline packages/backend/src/index.ts:99]: No need to merge the schemas afaik, this is just if you want to have a nice to display JSONSchema blob that isn't a bunch of separate small schemas

Comment (OrkoHunter) [reply packages/backend/src/index.ts:99]: Right, we need to get all the keys with visibility "secret" and so we thought merging is the first step. But we can obviously loop over all the individual schemas and do that. 😅

Comment (Rugvip) [reply packages/backend/src/index.ts:99]: All of that logic is already in the schema though, and you're not really interested in the keys at all, only the values. Something like `schema.process(config, {visibility: ['secret']})` should give you back a nested object with all secret values

**Comment by OrkoHunter** [comment]
Too draft this one. Will close this :)
Ref: https://github.com/backstage/backstage/pull/7156
cc @HHogg @ainhoaL

**Pull request closed without merging**